astropy/units/format/cds.py

```diff
@@ -157,11 +169,15 @@ def p_main(p):
             '''
             main : factor combined_units
                  | combined_units
+                 | OPEN_BRACKET combined_units CLOSE_BRACKET
                  | factor
             '''
             from astropy.units.core import Unit
+            from astropy.units import dex
             if len(p) == 3:
                 p[0] = Unit(p[1] * p[2])
+            elif len(p) == 4:
+                p[0] = dex(Unit(p[2]))
```

Comment:
It doesn't matter too much but I think this can just be `p[0] = dex(p[2])` (`combined_unit` is guaranteed to be a unit instance already).